addons/project/static/src/project_sharing/chatter/composer_patch.js

```diff
@@ -16,6 +16,6 @@ patch(Composer.prototype, {
     },
 
     get allowUpload() {
-        return this.thread?.id && super.allowUpload;
+        return (this.thread?.id || this.message) && super.allowUpload;
```

Comment:
I suppose this is the same with `isSendButtonDisabled` above?

From my understanding of code, both should be false on non-genuine thread (negative or 0 id).
Best to put
```js
if (this.thread.id && !this.thread.id) {
     return false;
}
```

So that we don't have to enumerate all contexts of composer. Just the link composer / thread is relevant for this override.